### PR TITLE
converts console.log logging to winston logging

### PR DESCRIPTION
**Pull request by ablwr**
This also improves logging around API responses.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
+  transports: [
+    new winston.transports.Console({
+      handleExceptions: true,
+      json: true
+    })
+  ],
+  exitOnError: false
+});
+
+logger.log('info', {'message': 'Loading Discovery Poster'})
```
Comment (kfriedman) [inline index.js]: Would it be cleaner to use `logger.info`?

**Comment by kfriedman** [comment]
Looks good! Couple of overall questions:
- Would it be better to use `logger.info` and `logger.error` conventions? I think that might be the more common syntax when using a logger like Winston.
- Can you confirm that the JSON is outputting as single-line and not multi-line? I think maybe we need to add/use the `stringify` parameter? I don't remember.

File: index.js [hunk 2]
```diff
@@ -67,26 +76,28 @@ exports.kinesisHandler = function (records, context, callback) {
...
+          CACHE['accessToken'] = null
         }
-      } else {
-        callback(null, 'POST Success')
+        logger.log('error', {'message': 'POST Error! ' + JSON.stringify(error)})
+        callback(new Error(error))
+        return
+      }
+      if (body.errors && body.errors.length) {
+        logger.log('info', {'message': 'Data error: ' + JSON.stringify(body.errors)})
```
Comment (kfriedman) [inline index.js]: I think there might be a way to pass Winston an object and just have it handle it... and not need to do `JSON.stringify`.

**Comment by ablwr** [comment]
> Would it be better to use logger.info and logger.error conventions? I think that might be the more common syntax when using a logger like Winston.

Sure, I can do it that way. Chose the more neutral option since we don't currently but may eventually have a framework+schema to follow, for consistency. 

> Can you confirm that the JSON is outputting as single-line and not multi-line? I think maybe we need to add/use the stringify parameter? I don't remember.

I didn't want to deploy before PR is approved, since these is now moving pieces in the pipeline. Locally, it was multi-line. I updated it to return one-line. Because of using stringify in the config, I was able to eliminate it from the JSON (your third comment).

**Comment by kfriedman** [comment]
Look great. Thanks Ashley!

**Pull request merged**